### ISSUE TITLE
Request: separate-compilation option

### ISSUE THREAD

**Issue by cxw42** [ToruNiina/toml11#209]
Would it be possible to add an option that separately compiles the parser and some of the other `detail` code that callers don't need?  That would speed up compilation quite a bit.  Thanks for considering this request!

Example of how this is done in another library: <https://github.com/gulrak/filesystem#using-it-as-forwarding-implementation-header>

**Comment by cxw42** [comment]
As an interim step, I was able to write a forwarding header and function using the default template arguments.  Here it is in case it helps anyone else:

**toml-wrapper.hpp**
```
#include <unordered_map>
#include <vector>

#include "toml/get.hpp"

toml::basic_value<toml::preserve_comments, std::unordered_map, std::vector>
parseTomlFile(std::string fname);
```
**toml-wrapper.cpp**
```
#include "toml.hpp"
#include "toml-wrapper.hpp"

toml::basic_value<toml::preserve_comments, std::unordered_map, std::vector>
parseTomlFile(std::string fname)
{
    return toml::parse(fname);
}
```

**Comment by ToruNiina** [comment]
I have just released v4.0.0 and now this feature is implemented.

By setting `cmake -DTOML11_PRECOMPILE=ON` or `#define TOML11_COMPILE_SOURCES`, you can compile `src/*.cpp` and it makes compilation ~2x faster.

**Comment by cxw42** [comment]
Much appreciated!  I can see the difference in my compile times!
